CDP4Common/Types/OrderedItemList.cs

```diff
@@ -574,5 +539,61 @@ private void Add(long sortKey, T item)
             this.sortedItems.Add(sortKey, item);
             this.highestKey = Math.Max(sortKey, this.highestKey);
         }
+
+        /// <summary>
+        /// Validates a value for being null.
+        /// </summary>
+        /// <param name="value">
+        /// The value to validate.
+        /// </param>
+        /// <param name="message">
+        /// The message to pass to an exception if validation not passes.
+        /// </param>
+        /// <exception cref="ArgumentNullException">
+        /// Thrown when <see paramref="value"/> is null.
+        /// </exception>
+        private void ValidateValueForNull(object value, string message = "")

```

Comment:
If you make the default value for message "An item cannot be null", you can remove it from 3 other places and it will be easier to reuse it for future changes.